### ISSUE TITLE
Good method to support "daemonless" usage in REST API

### ISSUE THREAD

**Issue by ids1024** [containers/podman#7210]
One of the features Podman advertises is being "daemonless". And I suppose it is perfectly usable without a daemon through the CLI or Go library.

But when using Podman from a language other than Go, probably the best way to interact with it is over the REST API (or the varlink API that is being removed). So [python-podman](https://github.com/containers/python-podman) uses that. But it seems this API requires a podman daemon to already be running, bound to a socket at a known path.

Unless I'm missing something, wouldn't it be desirable to have a good method to spawn a temporary podman daemon for use in bindings like `python-podman`? Perhaps if `podman system service` had an argument that makes it generate a socket path with something roughly equivalent to `mktemp /tmp/podmanXXXX.sock`, and print that to stdout. Then bindings in another language could invoke that and connect to the socket, securely and without relying on a daemon already running.

Does this seem like a reasonable use case and solution? Or is there a better way to do things? I guess from a certain perspective it would be better to make direct calls to libpod, in which case a C wrapper would probably be necessary to make it useable from arbitrary languages...

TLDR: How should bindings in other languages, like Python, allow using podman without having a daemon running already?

**Comment by TomSweeneyRedHat** [comment]
@jwhonce PTAL

**Comment by jwhonce** [comment]
@ids1024 We provide unit files for podman.service and podman.socket to allow podman to leverage the systemd socket activation protocol.

You may use `podman system connection ...` to configuration connections to the service running podman, if you are not comfortable writing complex URL's.

If you have questions IRC channel #podman on freenode is where we handle support.

**Comment by ids1024** [comment]
Hm. Yeah, I guess if it provides a user `.socket` file, that's probably a satisfactory solution.
